op-deployer/pkg/deployer/opcm/l2genesis2_test.go

```diff
@@ -0,0 +1,136 @@
+					L1FeeVaultWithdrawalNetwork:              genesis.WithdrawalNetwork("local"),
+					SequencerFeeVaultWithdrawalNetwork:       genesis.WithdrawalNetwork("local"),
+				},
+			}
+			initCfg.UpgradeScheduleDeployConfig.ActivateForkAtGenesis(rollup.Isthmus)
+
+			l1Deps := L1Deployments{
+				L1CrossDomainMessengerProxy: common.Address{'M'},
+				L1StandardBridgeProxy:       common.Address{'B'},
+				L1ERC721BridgeProxy:         common.Address{'E'},
+			}
+
+			newInput := L2Genesis2Input{
+				L1ChainID:                                new(big.Int).SetUint64(initCfg.L1ChainID),
+				L2ChainID:                                new(big.Int).SetUint64(initCfg.L2ChainID),
+				L1CrossDomainMessengerProxy:              l1Deps.L1CrossDomainMessengerProxy,
+				L1StandardBridgeProxy:                    l1Deps.L1StandardBridgeProxy,
+				L1ERC721BridgeProxy:                      l1Deps.L1ERC721BridgeProxy,
+				L2ProxyAdminOwner:                        initCfg.OwnershipDeployConfig.ProxyAdminOwner,
+				SequencerFeeVaultRecipient:               initCfg.L2VaultsDeployConfig.SequencerFeeVaultRecipient,
+				SequencerFeeVaultMinimumWithdrawalAmount: (*big.Int)(initCfg.L2VaultsDeployConfig.SequencerFeeVaultMinimumWithdrawalAmount),
+				SequencerFeeVaultWithdrawalNetwork:       big.NewInt(int64(initCfg.L2VaultsDeployConfig.SequencerFeeVaultWithdrawalNetwork.ToUint8())),
+				BaseFeeVaultRecipient:                    initCfg.L2VaultsDeployConfig.BaseFeeVaultRecipient,
+				BaseFeeVaultMinimumWithdrawalAmount:      (*big.Int)(initCfg.L2VaultsDeployConfig.BaseFeeVaultMinimumWithdrawalAmount),
+				BaseFeeVaultWithdrawalNetwork:            big.NewInt(int64(initCfg.L2VaultsDeployConfig.BaseFeeVaultWithdrawalNetwork.ToUint8())),
+				L1FeeVaultRecipient:                      initCfg.L2VaultsDeployConfig.L1FeeVaultRecipient,
+				L1FeeVaultMinimumWithdrawalAmount:        (*big.Int)(initCfg.L2VaultsDeployConfig.L1FeeVaultMinimumWithdrawalAmount),
+				L1FeeVaultWithdrawalNetwork:              big.NewInt(int64(initCfg.L2VaultsDeployConfig.L1FeeVaultWithdrawalNetwork.ToUint8())),
+				GovernanceTokenOwner:                     initCfg.GovernanceDeployConfig.GovernanceTokenOwner,
+				Fork:                                     big.NewInt(6), // corresponds to isthmus
+				UseInterop:                               false,
+				EnableGovernance:                         true,
+				FundDevAccounts:                          true,
+			}
+
+			tt.mutator(&initCfg, &newInput)
+
+			hostOld := createTestHost(t)
+			require.NoError(t, L2Genesis(hostOld, &L2GenesisInput{
+				L1Deployments: l1Deps,
+				L2Config:      initCfg,
+			}))
+
+			hostNew := createTestHost(t)
+			script, err := NewL2Genesis2Script(hostNew)
+			require.NoError(t, err)
+
+			require.NoError(t, script.Run(newInput))
+
+			dumpOld, err := hostOld.StateDump()
+			require.NoError(t, err)
+			dumpNew, err := hostNew.StateDump()
+			require.NoError(t, err)
```

Comment:
More of a comment for my own understanding. 

I see that [`writeGenesisAlocs`](https://github.com/ethereum-optimism/optimism/blob/cdd5e5df8484eb009fd63d415b93f6157518c7c6/packages/contracts-bedrock/scripts/L2Genesis.s.sol#L648) is not included in the new L2Genesis2.sol, and that this code works around that by handling the state dump. AFAICT op-deployer [relies on the script](https://github.com/ethereum-optimism/optimism/blob/cdd5e5df8484eb009fd63d415b93f6157518c7c6/op-deployer/pkg/deployer/opcm/l2genesis.go#L56-L67) to do the dumping.

Will this not create an issue with attempts to fully remove the original L2Genesis.sol, or is the plan to move the state dump into the go code like this test?

Reply:
"State dump" is a bit of a misnomer here. Nothing is actually dumped to disk. Instead, `StateDump()` returns a copy of the in-memory state after the script runs. We can safely remove the `writeGenesisAllocs` method even without this PR; it's currently unused since the tests and chain ops tooling both use the in-memory state dump as shown here.